### ISSUE TITLE
MongoCustomConversions are registered as bean but conversions are not triggered

### ISSUE THREAD

**Issue by devsteff** [bwaldvogel/mongo-java-server#176]
We switched from Spring Boot version 2.2.4 to version 2.3.4. With the new Mongo drivers from Springboot starter some of our tests failed because the registered conversion classes are not triggerd within save or find.

This is how we did the converter registration before:
`	@Override
	@Bean
	public CustomConversions customConversions() {
		List<Converter<?, ?>> converters = new ArrayList<>();
		converters.add( new BenutzerTypReadConverter() );
		converters.add( new BenutzerTypWriteConverter() );
		converters.add( new BenutzerPermissionTypReadConverter() );
		converters.add( new BenutzerPermissionTypWriteConverter() );
		converters.add( new ClientTypReadConverter() );
		converters.add( new ClientTypWriteConverter() );
		return new CustomConversions( converters );
	}
`

And this is the new registration. 
`    @Bean
    public MongoCustomConversions mongoCustomConversions() {
        List<Converter> converters = new ArrayList<>(6);
        converters.add( new BenutzerTypReadConverter() );
        converters.add( new BenutzerTypWriteConverter() );
        converters.add( new BenutzerPermissionTypReadConverter() );
        converters.add( new BenutzerPermissionTypWriteConverter() );
        converters.add( new ClientTypReadConverter() );
        converters.add( new ClientTypWriteConverter() );
        return new MongoCustomConversions(converters);
    }
`

As you can see, the Bean class is different due to API changes. The Converter classes are unchanged. This change works perfectly with the real MongoDB, but not with the bwaldvogel mongo-java-server in Memory DB. I've tried with version 1.24.0 (the former used version) and also with the latest 1.36.0. The bean ist created but for me it seems, that it is not used by the mongo-java-server. Converted methods are not called. When I remove the Converter registration in my old Springboot version, I receive exactly the same error behavior and the same tests fail.

**Comment by bwaldvogel** [comment]
Unfortunately this bug reports does not help me to reproduce your issue.
Do you know if there is a specific query that either fails with `mongo-java-server` or returns the wrong result?

**Comment by abobwhite** [comment]
I too am seeing this issue with the latest 1.38.0 and using Spring Boot 2.4.5 (and spring data mongo).

I am writing a converter for `ZonedDateTime` since there is no codec for it out-of-the-box. Every time I invoke `repository.save(document)`, I get an error: `org.bson.coedecs.configuration.CodecConfigurationException: Can't find a codec for class java.time.ZonedDateTime`.

```
@Configuration
public class MongoConfig {
    @Bean
    MongoCustomConversions mongoCustomConversions() {
        return new MongoCustomConversions(Lists.newArrayList(
            ZonedDateTimeToStringConverter.INSTANCE,
            StringToZonedDateTimeConverter.INSTANCE
        ));
    }
}

```

The same result if I change the bean type to `CustomConversions` instead of `MongoCustomConversions`....

**Comment by abobwhite** [comment]
Actually I found my issue. For the in-memory configuration with mongo-java-server, I was creating the `MongoTemplate` bean manually but not providing it the `MappingMongoConverter` which contained the `customConversions` created as a `MongoCustomConversions`. After providing the `MappingMongoConverter` to my in-memory config to the `MongoTemplate`, the error went away. 

TL;DR: PEBKAC

**Comment by TamaraTacic** [comment]
are there any news on this one? I'm suspecting a similar issue on spring boot 2.5.4

**Comment by msttmms** [comment]
hello, are there any updates? I'm suspecting a similar issue with spring boot 2.5.4?

**Comment by bwaldvogel** [comment]
Actually I forgot to close this issue. @abobwhite wrote in [his last comment](https://github.com/bwaldvogel/mongo-java-server/issues/176#issuecomment-854976507) that the "PEBKAC" ;)

If you found a similar problem, feel free to create a new issue with a small reproducer if you think there’s something that needs to be changed in `mongo-java-server`.

**Comment by huybn5776** [comment]
In my case, the solution is remove `MongoTemplate` bean which is copy from example, if `MappingMongoConverter` is already provided elsewhere.
```java
public class MongoTestServerConfiguration {
  - @Bean
  - public MongoTemplate mongoTemplate(MongoDbFactory mongoDbFactory) {
  -     return new MongoTemplate(mongoDbFactory);
  - }
}
```
